src/realm/object-store/sync/sync_session.cpp

```diff
@@ -191,6 +197,11 @@ void SyncSession::do_become_inactive(util::CheckedUniqueLock lock, Status status
     if (m_sync_manager) {
         m_sync_manager->unregister_session(m_db->get_path());
     }
+    if (m_needs_subscription_store_updated) {
```

Comment:
I'm not sure why we need to do this here?

Reply:
I think the idea is to add/remove the subscription store when the session is closed

Reply:
when m_session is closed because a migrate error message was received from the server? Don't we bypass this function after a migration by calling restart_session here? Or is there some case where the user could close their session that we need to handle?

Reply:
the session is closed in download_fresh_realm (once the client reset handling is implemented)